### DISCUSSION TITLE
一些功能测试及建议

### DISCUSSION THREAD

**Discussion by yylqing** [usememos/memos#28] [Ideas]
1. 选择Tag之后，直接空格再输入内容，Tag可以被识别。如果输入Tag后回车再输入，Tag将不会被识别
2. 在代码模块里，当用# 号作为注释时，也会被识别为Tag。【这一点看下您的一个设计思路是否会全功能支持代码块；对于较长的代码块看是否可以实现折叠显示】
![image](https://user-images.githubusercontent.com/18625743/149050763-94c1ee6b-83b0-443a-a468-03c973445c98.png)
3. 当前数据库的密码是明文存储的，作为自用倒是无所谓，后续是否会做密码加密的功能
4. 是否可以添加一个“开放注册”的选项。我理解的当前是这个可能是个人自用，就不希望开放注册功能。当然这会带来一个问题，如何添加自己的用户，这样就增加了部署成本（需要有数据库操作经验）；可以有个选项控制注册用户的总数或者是程序启动时的一个传参
5. 后续有无预留插件的想法，以便自由添加一些功能模块

非常感谢作者的付出！

**Comment by boojack** [comment]
1. 目前 tag 的格式为：`# tagname ` 或者 `#tagname `，tagname 后面有一个空格，**后续将只支持 `#tagname `**；
2. 目前仅支持解析成代码块，由于和 tag 的格式冲突，建议不要使用 `#`；
3. 可以做；
4. 会增加一个设置表来管理站点配置信息，包括：是否开放注册等；（这也引发了我一个想法：通过一个站点来可视化生成/配置用户的 .db 文件）
5. 目前正在做减法/抽象，只留核心功能，其他功能将以插件的形式使用，包括 2 提到的代码块显示。

**Comment by yylqing** [comment]
2. 好的。结合5来说，做成插件的形式，展示的时候Markdown解析还有待完善，代码块中的 * 也会被解析；不过这个是后续插件的内容
4. 增加一个管理员页面，就可以将插件管理、站点配置、用户管理放进去；这样可以保持系统的前台只有核心的功能，最纯粹的记录、写作页面

6. 目前发现有二级Tag的情况，点击一级Tag不能筛选出相应的memo

**Comment by rebron1900** [comment]
基础功能就我目前使用来说都基本完善了，它具备了一个碎片化笔记的所有功能了。

关于拓展功能，我就自己的需要进行了设想，大概内容如下。

1. 提供一些API，如：创建memos，获取指定memos的json（通过，tag，时间，等参数）
2. tag排序、tag置顶、tag图标、tag颜色
3. 更丰富的编辑器
4. 拓展功能，如：图床设置、特殊化的memos卡片（**天气、视频、音频、网址等等**，这是我觉得flomo目前不行的地方。既然是一个碎片化的工具，那要就能装下各种碎片。）、
5. 希望标签像flomo那样独立显示，因为标签和memos内容上没有关联，只作为分类标识存在。

**Comment by yylqing** [comment]
7. 当前的搜索是大小写匹配的，是否可以改成模糊搜索，实际使用中不需要大小写严格匹配
8. 能否添加按时间倒序/顺序的排列方式，目前是最新的在最上面，是否可以添加按钮按时间轴顺序查看
9. 如上面 @rebron1900 提到的4，作为碎片化的工具，卡片的类型可以更丰富，包括**地理位置**这样的卡片

**Comment by boojack** [comment]
**丰富的卡片**是一个不错的idea（有个小彩蛋不知各位发现了没，点击热力图上面的 DAY，这算是一个 Daily Timeline 的卡片形式

**Comment by rebron1900** [comment]
@boojack 这个彩蛋老早就发现了，上次还说能不能根据筛选后的数据进行展示。

关于这个丰富的卡片，是因为我使用的博客系统 [ghost](https://github.com/TryGhost/Ghost) 近一年的更新都在丰富这个卡片的使用，我觉得很酷，也很实用。
至于业务流程、展示等方面是需要讨论的问题。


[https://www.helloimg.com/images/2022/01/12/G7cXFq.png](https://www.helloimg.com/images/2022/01/12/G7cXFq.png)

**Comment by rebron1900** [comment]
10. 标签的修改。现在想修改某个标签信息需要把这个标签下的所有memos手动修改才可以，有点麻烦。

**Comment by gsgqgyq** [comment]
希望可以有那年今日和随机回顾的插件

**Comment by boojack** [comment]
> 希望可以有那年今日和随机回顾的插件

一些小玩具功能会加在“快速检索”里，比如：随机漫步、每日回顾。

**Comment by yylqing** [comment]
1. 目前发现，页面数据是一次性加载的，缺少分页功能。后续可以考虑加上分页加载，以免数据过多时影响到加载速度
2. 当前的图片样式还有待完善；多图时，可以限制一个图片的宽度然后剪裁展示以达到统一（现在展示也挺好看了，只是为了更美观）
![image](https://user-images.githubusercontent.com/18625743/150052290-aecb320a-7d5f-444f-8850-cc8e96b307ce.png)

**Comment by yylqing** [comment]
对于图片预览，目前点开是单图片预览，无法左右滑动切换图片，关闭预览需要点击关闭按钮，无法点击空白处退出预览

**Comment by rebron1900** [comment]
接上面
图片预览界面的图标意思不明确，我第一次使用的时候没明白各个按钮的作用

**Comment by yylqing** [comment]
标签中包含中英文的时候，展示存在问题，标签是可以正常识别到的。
![image](https://user-images.githubusercontent.com/18625743/151091564-2c8c62ca-79bb-4444-864d-8ea6a901e1d3.png)

**Comment by boojack** [comment]
> 标签中包含中英文的时候，展示存在问题，标签是可以正常识别到的。 ![image](https://user-images.githubusercontent.com/18625743/151091564-2c8c62ca-79bb-4444-864d-8ea6a901e1d3.png)

Fixed in https://github.com/justmemos/memos/commit/374721fa485dc0aafb1f1561ec96c82581a892cc
